### PR TITLE
reload aead when encountering CIF chacha20poly1305 error

### PR DESCRIPTION
**Pull request by yjst2012**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-4249

### What this PR does / why we need it:

reload aead when encountering CIF chacha20poly1305 error

### Test plan for issue:

manually modified code to test it:
INFO[2024-03-12T13:52:06+13:00]pkg/cluster/graph/manager.go:60 graph.(*manager).LoadPersisted() Jeff failed to load persisted with current aead &{0xc00065d080 [0xc00024e6e0 0xc0002b70c0]}  client_principal_name= client_request_id= component=backend correlation_id= request_id=ff0aad89-b360-48ce-bfe1-607fff22a651 resource_group=v4-australiaeast resource_id=/subscriptions/fe16a035-e540-4ab7-80d9-373fa9a3d6ae/resourcegroups/v4-australiaeast/providers/microsoft.redhatopenshift/openshiftclusters/jeff-412x resource_name=jeff-412x subscription_id=fe16a035-e540-4ab7-80d9-373fa9a3d6ae version=4.12.25
INFO[2024-03-12T13:52:06+13:00]pkg/cluster/graph/manager.go:73 graph.(*manager).reloadAead() Jeff key changed, re-freshing key & recreating graph...  client_principal_name= client_request_id= component=backend correlation_id= request_id=ff0aad89-b360-48ce-bfe1-607fff22a651 resource_group=v4-australiaeast resource_id=/subscriptions/fe16a035-e540-4ab7-80d9-373fa9a3d6ae/resourcegroups/v4-australiaeast/providers/microsoft.redhatopenshift/openshiftclusters/jeff-412x resource_name=jeff-412x subscription_id=fe16a035-e540-4ab7-80d9-373fa9a3d6ae version=4.12.25
INFO[2024-03-12T13:52:07+13:00]pkg/cluster/graph/manager.go:68 graph.(*manager).LoadPersisted() Jeff re-try load persisted with refreshed aead &{0xc007ed2220 [0xc007a2e600 0xc007a2e8c0]}  client_principal_name= client_request_id= component=backend correlation_id= request_id=ff0aad89-b360-48ce-bfe1-607fff22a651 resource_group=v4-australiaeast resource_id=/subscriptions/fe16a035-e540-4ab7-80d9-373fa9a3d6ae/resourcegroups/v4-australiaeast/providers/microsoft.redhatopenshift/openshiftclusters/jeff-412x resource_name=jeff-412x subscription_id=fe16a035-e540-4ab7-80d9-373fa9a3d6ae version=4.12.25
INFO[2024-03-12T13:52:07+13:00]pkg/cluster/graph/manager.go:84 graph.(*manager).loadPersisted() Jeff load persisted graph                     client_principal_name= client_request_id= component=backend correlation_id= request_id=ff0aad89-b360-48ce-bfe1-607fff22a651 resource_group=v4-australiaeast resource_id=/subscriptions/fe16a035-e540-4ab7-80d9-373fa9a3d6ae/resourcegroups/v4-australiaeast/providers/microsoft.redhatopenshift/openshiftclusters/jeff-412x resource_name=jeff-412x subscription_id=fe16a035-e540-4ab7-80d9-373fa9a3d6ae version=4.12.25

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

File: pkg/cluster/graph/manager_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		},
+		{
+			name: "loadPersisted returns an error other than the chacha20poly1305 one",
+			mocks: func(storage *mock_storage.MockManager, env *mock_env.MockInterface, kv *mock_keyvault.MockManager) {
+				storage.EXPECT().BlobService(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(&azstorage.BlobStorageClient{}, errors.New("general error"))
+			},
+			wantErr: "general error",
+		},
+		{
+			name: "loadPersisted returns a chacha20poly1305 error",
```
Comment (cadenmarchese) [inline pkg/cluster/graph/manager_test.go:44]: Should we add one more test case for no error / `wantErr: nil`?

**Review by cadenmarchese — COMMENTED** [review]
Retry logic looks good to me!

File: pkg/cluster/graph/manager_test.go (hunk 1 above)
Comment (yjst2012) [reply pkg/cluster/graph/manager_test.go:44]: the problem is that some structs like BlobStorageClient and azstorage are under vendor folder that do not have a mock interface, hence can not be properly controlled to craft a nil error case :(, guess this is why there is no test cases for graph and some cluster files before.
these few test cases are mostly covering the changes in the pr, the content of loadPersisted() are not touched this time, so it should work as before.

**Review by cadenmarchese — APPROVED** [review]

**Comment by yjst2012** [comment]
> I have a question about it. Is there no possibility that decrypting a graph encrypted with an old key fail because the RP reloads the key to the new one?
RP tries the old key first and reloads the new key only if it fails, so guess this wont happen?
 
> It seems that LoadPersisted() is also used when upgrading the cluster.
you meant updateClusterData() right? it looks like only get called by Install() in the cluster provisioning state, not upgrading.
